src/main/java/org/datadog/jmxfetch/util/ByteArraySearcher.java

```diff
@@ -0,0 +1,47 @@
+package org.datadog.jmxfetch.util;
+
+public final class ByteArraySearcher {
```

Comment:
interesting matching technique - I think it would be worth adding a small description of the algorithm used here

Reply:
This is the literal version of the [bitap algorithm](https://en.wikipedia.org/wiki/Bitap_algorithm) which is the only one I know which takes less space to write than the naive quadratic algorithm used in Guava.